sdks/java/core/src/test/java/org/apache/beam/sdk/options/ProxyInvocationHandlerTest.java

```diff
@@ -683,10 +722,233 @@ public void testJsonConversionOfSerializableWithMetadataProperty() throws Except
     assertEquals("TestString", options2.getValue().getValue());
   }
 
+  @Test
+  public void testDisplayDataItemProperties() {
+    PipelineOptions options = PipelineOptionsFactory.create();
+    options.setTempLocation("myTemp");
+    DisplayData displayData = DisplayData.from(options);
+
+    assertThat(displayData, hasDisplayItem(allOf(
+        hasKey("tempLocation"),
+        hasType(DisplayData.Type.STRING),
+        hasValue("myTemp"),
+        hasNamespace(PipelineOptions.class)
+    )));
+  }
+
+  @Test
+  public void testDisplayDataTypes() {
+    Instant now = Instant.now();
+
+    TypedOptions options = PipelineOptionsFactory.as(TypedOptions.class);
+    options.setInteger(1234);
+    options.setTimestamp(now);
+    options.setJavaClass(ProxyInvocationHandlerTest.class);
+    options.setObject(new Serializable() {
```

Comment:
In theory, the string representation seems like a better fit as display data is for human comprehension, rather than data serialization. If we notice that much of this data doesn't have good string representations, we should consider making the switch.